taskvine/src/bindings/python3/ndcctools/taskvine/dask_executor.py

```diff
@@ -273,6 +281,7 @@ def _dask_execute(self, dsk, keys):
                         print(f"{t.key} ran on {t.hostname}")
 
                     if t.successful():
+                        self.category_execution_time[t.category].append(t.execution_time)
```

Comment:
I don't think you want to keep all the execution times around in that list, as it looks like a potential memory leak?
Probably you can do a rolling average (a variable to keep the sum of the execution times and a variable to keep how many tasks of that category.)